automerge-backend/src/object_store.rs

```diff
@@ -52,6 +60,39 @@ impl ListState {
         }
     }
 
+    fn generate_diffs(&self) ->  Vec<Diff> {
+        let mut after = 0;
+        let ops_in_order = list_ops_in_order(&self.operations_by_elemid, &self.following).ok().unwrap_or(vec![]);
```

Comment:
Here it would be better to use `unwrap_or_else(Vec::new)`, this avoids allocating unless there actually is an unwrap needed.